src/components/sidebar/ContextAwareSidebar.tsx

```diff
@@ -275,6 +317,88 @@ class ContextAwareSidebar
       </SidebarGroup>
     );
 
+    const getRefGuidFromRef = (ref: Edge) => {
+      const id = stripId(ref.sourceId);
+
+      return Maybe.maybe(course.resourcesById.get(id)).caseOf({
+        just: resource => resource.guid,
+        nothing: () => '',
+      });
+    };
+
+    const stripId = (id: string) => {
+      const splits = id.split(':');
+      if (splits.length === 3) {
+        return splits[2];
+      }
+    };
+
+    const getRefTitleFromRef = (ref: Edge) => {
+      const id = stripId(ref.sourceId);
+
+      return Maybe.maybe(course.resourcesById.get(id)).caseOf({
+        just: resource => resource.title,
+        nothing: () => '[Error loading page title]',
+      });
+    };
+
+    const orgGuid = selectedOrganization.caseOf({
+      just: (organization) => {
+        return organization.guid;
+      },
+      nothing: () => '',
+    });
+
+    /* This const is responsible for constructing and displaying a list of clickable
+     * reference links
+     */
+    const referenceLocations = (
+      <SidebarGroup label="Referenced Locations">
+        <SidebarRow>
+          <div className="page-list">
+            {resourceRefs.caseOf({
+            /* KYLE-1936 finding source of resourceRefs*/
+              just: (refs) => {
+                return refs.length > 0
+                ? (
+                  <div className="container">
+
+                    {
+                      refs.map(ref => (
+                      <div key={ref.guid} className="ref-thing">
+                        <a href="#" onClick = {(event) => {
+                          event.preventDefault();
+                          // if link is to org, just switch org and stay on current page
+                          if (ref.sourceType === 'x-oli-organization') {
+                            viewDocument(resource.id, course.idvers,
+                                         Maybe.maybe(stripId(ref.sourceId)));
+                          } else {
+                            viewDocument(stripId(ref.sourceId), course.idvers, Maybe.nothing());
+                          }
+                        }
+                        }>
+                          <span style={{ width: 26, textAlign: 'center', marginRight: 5 }}>
+                            {
+                              getNameAndIconByType(ref.sourceType).icon}
+                            </span>
+                            {getRefTitleFromRef(ref)}
+                        </a>
+                      </div>
+                    ))}
+                  </div>
+                ) : (
+                  <div>No references found</div>
+                );
+              },
+              nothing: () => <div>Error loading</div>,
```

Comment:
I don't think that "Error loading" is appropriate to display here.  resourceRefs is Nothing by default - which means until the network request completes "Error loading" is what will be displayed. You are seeing this in your UI, but it is probably completing so fast that you never notice it.  We should only display "Error loading" when the network request actually has failed.

Reply:
Now cases on new state variable to display error when necessary, loading message otherwise